File: packages/dry/fixtures/python-duplicate-pair/src/invoice.py
def compute_order_total(lines: list[dict[str, float]]) -> float:
    goods = 0.0
    taxes = 0.0
    for line in lines:
        quantity = line["quantity"]
        unit_price = line["unit_price"]
        tax_rate = line["tax_rate"]
        extended = quantity * unit_price
        goods += extended
        taxes += extended * tax_rate
        if quantity >= 25:
            goods -= extended * 0.08
        elif quantity >= 10:
            goods -= extended * 0.03
    freight = 0.0 if goods > 250 else 8.5 if goods > 100 else 15.0
    coupon = 25.0 if goods > 400 else 0.0
    return goods + taxes + freight - coupon

File: packages/dry/fixtures/semantic-pair/src/invoice.ts
export function computeOrderTotal(
  lines: ReadonlyArray<{ quantity: number; unitPrice: number; taxRate: number }>,
): number {
  let goods = 0;
  let taxes = 0;
  for (const line of lines) {
    const extended = line.quantity * line.unitPrice;
    goods += extended;
    taxes += extended * line.taxRate;
    if (line.quantity >= 25) {
      goods -= extended * 0.08;
    } else if (line.quantity >= 10) {
      goods -= extended * 0.03;
    }
  }
  const freight = goods > 250 ? 0 : goods > 100 ? 8.5 : 15;
  const coupon = goods > 400 ? 25 : 0;
  return goods + taxes + freight - coupon;
}
